vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c

```diff
@@ -1291,9 +1291,11 @@ WIFIReturnCode_t WIFI_ConfigureAP( const WIFINetworkParams_t * const pxNetworkPa
         }
 
         /* ssid/password is required */
-        strlcpy((char *) &wifi_config.ap.ssid, pxNetworkParams->pcSSID, pxNetworkParams->ucSSIDLength);
+        // SSID can be a non NULL terminated string if ssid_len is specified. Hence memcpy is used.
+        memcpy((char *) &wifi_config.ap.ssid, pxNetworkParams->pcSSID, pxNetworkParams->ucSSIDLength);
+        wifi_config.ap.ssid_len = pxNetworkParams->ucSSIDLength;
         if (pxNetworkParams->xSecurity != eWiFiSecurityOpen) {
-            strlcpy((char *) &wifi_config.ap.password, pxNetworkParams->pcPassword, pxNetworkParams->ucPasswordLength);
+            strlcpy((char *) &wifi_config.ap.password, pxNetworkParams->pcPassword, pxNetworkParams->ucPasswordLength + 1);
```

Comment:
The copy operations of SSID ( using `memcpy` ) and password (using `strlcpy`) are inconsistent. Can they both use either `memcpy` or `strlcpy`?

Reply:
There is an option to have SSID as a non NULL terminated string and set the length in ssid_len (as specified in the [documentation](https://github.com/espressif/esp-afr-sdk/blob/ee3a985073c2474defb20ab353dd4be0dfd59956/components/esp32/include/esp_wifi_types.h#L213)).

In case of password, it is mandatory to have NULL terminated string. Hence, it is necessary to use `strlcpy` in case of password.

If `memcpy` is not used for SSID, the maximum length will be reduced to 31 bytes.

Reply:
Okay, thank you for the clarification. A comment to explicitly mentioning the reason for using `memcpy` to support 32 characters long SSID name could be helpful.